### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by WarGloom**
a little changes to compile on linux

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -59,6 +59,7 @@ add_boost_library(system)
 if (Boost_MAJOR_VERSION GREATER 0 AND Boost_MINOR_VERSION GREATER 47)
     add_boost_library(chrono)
 endif ()
+add_boost_library(regex)
```
Comment (taxilian) [inline CMakeLists.txt]: Why is this needed?

File: src/ScriptingCore/utf8_tools.cpp [hunk 2]
```diff
@@ -17,24 +17,39 @@ Copyright 2009 Dan Weatherford, Facebook inc
...
-#include <codecvt>
+#include <boost/locale/encoding_utf.hpp>
+#include <string>
+
+///#include <codecvt>
+//#include <locale>
 
 #include "utf8_tools.h"
 
-std::wstring_convert<std::codecvt_utf8<wchar_t>> utf8_conv;
```
Comment (taxilian) [inline src/ScriptingCore/utf8_tools.cpp]: The version here is using built-in C++11 routines to do the conversion; I'd rather not rely on the boost conversions unless there is a particularly strong reason to do so.

What is the reason you've changed this?

File: CMakeLists.txt (hunk 1 above)
Comment (WarGloom) [inline CMakeLists.txt]: without regex,  I got ld error: 
CMakeFiles/FireWyrmNativeMessageHost.dir/X11/PluginLoaderLin.cpp.o: In function `boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::assign(char const*, char const*, unsigned int)':
PluginLoaderLin.cpp:(.text._ZN5boost11basic_regexIcNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEE6assignEPKcS7_j[_ZN5boost11basic_regexIcNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEE6assignEPKcS7_j]+0x2a): undefined reference to `boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::do_assign(char const*, char const*, unsigned int)'
CMakeFiles/FireWyrmNativeMessageHost.dir/X11/PluginLoaderLin.cpp.o: In function `bool boost::regex_search<__gnu_cxx::__normal_iterator<char const*, std::string>, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::string> > >, char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >(__gnu_cxx::__normal_iterator<char const*, std::string>, __gnu_cxx::__normal_iterator<char const*, std::string>, boost::match_results<__gnu_cxx::__normal_iterator<char const*, std::string>, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::string> > > >&, boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, boost::regex_constants::_match_flags, __gnu_cxx::__normal_iterator<char const*, std::string>)':
PluginLoaderLin.cpp:(.text._ZN5boost12regex_searchIN9__gnu_cxx17__normal_iteratorIPKcSsEESaINS_9sub_matchIS5_EEEcNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEEEbT_SD_RNS_13match_resultsISD_T0_EERKNS_11basic_regexIT1_T2_EENS_15regex_constants12_match_flagsESD_[_ZN5boost12regex_searchIN9__gnu_cxx17__normal_iteratorIPKcSsEESaINS_9sub_matchIS5_EEEcNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEEEbT_SD_RNS_13match_resultsISD_T0_EERKNS_11basic_regexIT1_T2_EENS_15regex_constants12_match_flagsESD_]+0xab): undefined reference to `boost::re_detail::perl_matcher<__gnu_cxx::__normal_iterator<char const*, std::string>, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::string> > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::find()'
CMakeFiles/FireWyrmNativeMessageHost.dir/X11/PluginLoaderLin.cpp.o: In function `boost::re_detail::perl_matcher<__gnu_cxx::__normal_iterator<char const*, std::string>, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::string> > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::perl_matcher(__gnu_cxx::__normal_iterator<char const*, std::string>, __gnu_cxx::__normal_iterator<char const*, std::string>, boost::match_results<__gnu_cxx::__normal_iterator<char const*, std::string>, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::string> > > >&, boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, boost::regex_constants::_match_flags, __gnu_cxx::__normal_iterator<char const*, std::string>)':
PluginLoaderLin.cpp:(.text._ZN5boost9re_detail12perl_matcherIN9__gnu_cxx17__normal_iteratorIPKcSsEESaINS_9sub_matchIS6_EEENS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEEC2ES6_S6_RNS_13match_resultsIS6_S9_EERKNS_11basic_regexIcSD_EENS_15regex_constants12_match_flagsES6_[_ZN5boost9re_detail12perl_matcherIN9__gnu_cxx17__normal_iteratorIPKcSsEESaINS_9sub_matchIS6_EEENS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEEC5ES6_S6_RNS_13match_resultsIS6_S9_EERKNS_11basic_regexIcSD_EENS_15regex_constants12_match_flagsES6_]+0x116): undefined reference to `boost::re_detail::perl_matcher<__gnu_cxx::__normal_iterator<char const*, std::string>, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::string> > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::construct_init(boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, boost::regex_constants::_match_flags)'
collect2: error: ld returned 1 exit status
firebreath/NativeMessageHost/CMakeFiles/FireWyrmNativeMessageHost.dir/build.make:203: recipe for target 'firebreath/NativeMessageHost/FireWyrmNativeMessageHost' failed
make[2]: *** [firebreath/NativeMessageHost/FireWyrmNativeMessageHost] Error 1
CMakeFiles/Makefile2:707: recipe for target 'firebreath/NativeMessageHost/CMakeFiles/FireWyrmNativeMessageHost.dir/all' failed
make[1]: *** [firebreath/NativeMessageHost/CMakeFiles/FireWyrmNativeMessageHost.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2

File: src/ScriptingCore/utf8_tools.cpp (hunk 2 above)
Comment (WarGloom) [inline src/ScriptingCore/utf8_tools.cpp]: I got this compilation error
```bash
Building CXX object firebreath/ScriptingCore/CMakeFiles/ScriptingCore.dir/utf8_tools.cpp.o
cd /home/nikita/work/Projects/rucrypto/rtPluginPINPad/external/firebreath/build/firebreath/ScriptingCore && /usr/lib64/ccache/c++    -DFB_X11=1 -DXP_UNIX -fPIC -DBOOST_ALL_NO_LIB=1 -DUNICODE -D_UNICODE -DFB_X11=1 -DXP_UNIX -fPIC -DBOOST_ALL_NO_LIB=1 -DUNICODE -D_UNICODE -I/home/nikita/work/Projects/rucrypto/rtPluginPINPad/external/firebreath/src/ScriptingCore -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/freetype2 -I/home/nikita/work/Projects/rucrypto/rtPluginPINPad/external/firebreath/src/config -I/home/nikita/work/Projects/rucrypto/rtPluginPINPad/external/firebreath/src/PluginCore -I/home/nikita/work/Projects/rucrypto/rtPluginPINPad/external/firebreath/src/3rdParty/boost    -std=gnu++11 -o CMakeFiles/ScriptingCore.dir/utf8_tools.cpp.o -c /home/nikita/work/Projects/rucrypto/rtPluginPINPad/external/firebreath/src/ScriptingCore/utf8_tools.cpp
/home/nikita/work/Projects/rucrypto/rtPluginPINPad/external/firebreath/src/ScriptingCore/utf8_tools.cpp:21:19: fatal error: codecvt: No such file or directory
 #include <codecvt>
```
now I think there should be a better solution...
